pkg/api/types.go

```diff
@@ -1305,8 +1305,9 @@ type Endpoints struct {
 //     a: [ 10.10.1.1:8675, 10.10.2.2:8675 ],
 //     b: [ 10.10.1.1:309, 10.10.2.2:309 ]
 type EndpointSubset struct {
-	Addresses []EndpointAddress
-	Ports     []EndpointPort
+	Addresses         []EndpointAddress
+	NotReadyAddresses []EndpointAddress
```

Comment:
@smarterclayton Your response about splitting the subset does not seem to be in line with your code change. per your code-changes, it does seem that both arrays (Addresses and NotReadyAddresses) can be in the same subset, but one endpointAddress can only be in one or the other arrays.